src/cmd.rs

```diff
@@ -71,6 +71,59 @@ impl<'a, T: FromRedisValue> Iterator for Iter<'a, T> {
     }
 }
 
+use crate::aio::ConnectionLike as AsyncConnection;
+
+/// Represents a redis iterator that can be used with async connections.  
+#[cfg(feature = "aio")]
+pub struct AsyncIter<'a, T: FromRedisValue + 'a> {
+    batch: Vec<T>,
```

Comment:
Instead of reversing and storing a `Vec` here you could store `Vec`s `IntoIter` and call `next` instead of `pop` below here. Same could be done for `Iterator` variant.

Reply:
I'd rather not change existing functionality as a first-time contriubtor, so I think it's best to follow the pattern.